forage-android/src/main/java/com/joinforage/forage/android/core/telemetry/Metrics.kt

```diff
@@ -0,0 +1,204 @@
+package com.joinforage.forage.android.core.telemetry
+import com.joinforage.forage.android.VaultType
+
+internal object MetricsConstants {
+    const val PATH = "path"
+    const val METHOD = "method"
+    const val HTTP_STATUS = "http_status"
+    const val RESPONSE_TIME_MS = "response_time_ms"
+    const val ACTION = "action"
+    const val VAULT_TYPE = "vault_type"
+    const val RESPONSE_TYPE = "response_type"
+}
+
+internal enum class ActionType(val value: String) {
+    BALANCE("balance"),
+    CAPTURE("capture");
+
+    override fun toString(): String {
+        return value
+    }
+}
+
+internal enum class ResponseType(val value: String) {
+    /*
+    VAULT_RESPONSE_TIME response type refers to the response time from the VGS and BT submit
+    functions. The timer begins when a balance or capture request is submitted to VGS/BT
+    and ends when a response is received by the SDK.
+     */
+    VAULT_RESPONSE_TIME("vault_response_time"),
+
+    /*
+    CUSTOMER_PERCEIVED_RESPONSE_TIME type refers to the response time that a customer
+    experiences while executing a balance or capture action. There are multiple chained requests
+    that come from the client when executing a balance or capture action. The timer begins when the
+    first HTTP request is sent from the SDK and ends when the the SDK returns information back to
+    the user. Ex of a balance action:
+    Timer Begins -> [GET] EncryptionKey -> [GET] PaymentMethod -> [POST] to VGS/BT ->
+    [GET] Poll for Response -> [GET] PaymentMethod -> Timer Ends -> Return Balance
+     */
+    CUSTOMER_PERCEIVED_RESPONSE_TIME("customer_perceived_response_time");
+
+    override fun toString(): String {
+        return value
+    }
+}
```

Comment:
the vault log and the customer-perceived log will/can be re-used for tracking error rates (like I did for iOS SDK), meaning that these logs are not time-specific, so the `_time` suffix can be removed

can we do something like 

```kotlin
internal enum class EventName {
  VAULT_RESPONSE
  
  CUSTOMER_PERCEIVED_RESPONSE
}
```

@devinmorgan @miketikh